packages/block-library/src/gallery/edit.js

```diff
@@ -109,8 +109,10 @@ class GalleryEdit extends Component {
 	}
 
 	onSelectImages( images ) {
+		const { columns } = this.props.attributes;
 		this.setAttributes( {
 			images: images.map( ( image ) => pickRelevantMediaFiles( image ) ),
+			columns: Math.min( images.length, columns ),
```

Comment:
Maybe we should use `columns: columns ? Math.min( images.length, columns ) : columns,` here like we do in onRemoveImage. This ensures that we will not set the columns attribute if it was not set before.

Reply:
Whoa, I updated this to make that change but in trying to rebase my two commits into one I royally messed something up and commits from other people are showing up here. Since it's such a simple code change, Instead of trying to fix this pull request (and likely making it worse), I'm just going to close this and open a different, sorry for the inconvenience @jorgefilipecosta